fendermint/vm/interpreter/src/fvm/checkpoint.rs

```diff
@@ -55,17 +65,35 @@ where
     match should_create_checkpoint(gateway, state, height)? {
         None => Ok(None),
         Some(subnet_id) => {
-            // Get the current power table.
-            let power_table = get_power_table(client, height)
+            // Get the current power table from CometBFT.
+            // NB: Here we could also get it from the IPC Gateway.
+            let power_table = bft_power_table(client, height)
                 .await
                 .context("failed to get the power table")?;
 
-            // TODO #252: Take the next validator changes from the gateway.
-            let power_updates = PowerUpdates(Vec::new());
+            // Apply any validator set transitions.
+            let next_configuration_number = gateway
+                .apply_validator_changes(state)
+                .context("failed to apply validator changes")?;
+
+            // Figure out the power updates if there was some change in the configuration.
+            let power_updates = if next_configuration_number == 0 {
```

Comment:
By the way @adlrocha do you think it will cause any issues for Solidity if we send 0 in the checkpoint to show "nothing changed"?

Reply:
I don't think is a problem (that is what I am returning myself when applying the finality). What I am wondering now is where are we confirming the new membership in the parent when the bottom-up checkpoint is committed, I can't find anything [here](https://github.com/consensus-shipyard/ipc-solidity-actors/blob/f492531ae7d441b4bb00ca930f5d066956fadf4f/src/subnet/SubnetActorManagerFacet.sol#L40).

@cryptoAtwill, @dnkolegov, do you mind chiming in here. I am missing something or are we not confirming the changes in membership yet and we should? (I think it is the latter after reading the code, this should be an easy fix, but is a must :) )

Reply:
Thanks for the link @adlrocha , it indeed looks like the membership is not altered to reflect the new configuration number. We discussed that once that is done, we can't use `validateActiveQuorumSignature` any more and Denis updated the method you link to compare hashes, but the actual transition is missing :mag:

Reply:
@dnkolegov can you pick this one up, if not I think I can pick it up when I wrap my current PR (I am updating the contract bindings in `ipc-agent`): https://github.com/consensus-shipyard/ipc-agent/pull/334

Reply:
ok

Reply:
Here - https://github.com/consensus-shipyard/ipc-solidity-actors/blob/6322ae6a236859cedcc32d6ff7dabf4d709695e8/src/subnet/SubnetActorManagerFacet.sol#L73